catroid/src/main/java/org/catrobat/catroid/formulaeditor/FormulaElement.java

```diff
@@ -500,19 +500,24 @@ private Object interpretFunction(Functions function, Sprite sprite) {
 				break;
 			case RASPIDIGITAL:
 				RPiSocketConnection connection = RaspberryPiService.getInstance().connection;
-				int pin = doubleValueOfLeftChild.intValue();
-				try {
-					return connection.getPin(pin) ? 1d : 0d;
-				} catch (Exception e) {
-					Log.e(getClass().getSimpleName(), "RPi: exception during getPin: " + e);
+				if (doubleValueOfLeftChild != null) {
+					int pin = doubleValueOfLeftChild.intValue();
+					try {
+						return connection.getPin(pin) ? 1d : 0d;
+					} catch (Exception e) {
+						Log.e(getClass().getSimpleName(), "RPi: exception during getPin: " + e);
+					}
 				}
 				break;
 			case MULTI_FINGER_TOUCHED:
-				return TouchUtil.isFingerTouching(doubleValueOfLeftChild.intValue()) ? 1d : 0d;
+				return (doubleValueOfLeftChild != null && TouchUtil.isFingerTouching(doubleValueOfLeftChild.intValue
+						())) ? 1d : 0d;
 			case MULTI_FINGER_X:
-				return Double.valueOf(TouchUtil.getX(doubleValueOfLeftChild.intValue()));
+				return doubleValueOfLeftChild != null ? (double) TouchUtil.getX(doubleValueOfLeftChild
+						.intValue()) : 0d;
```

Comment:
can you put 0d to the top of this method as `final double defaultReturnValue = 0d` and use it everywhere. easier to read ;)